tooling/src/hypothesistooling/projects/hypothesispython.py

```diff
@@ -171,6 +179,17 @@ def build_distribution():
 
 def upload_distribution():
     tools.assert_can_release()
+
+    # This block ensures that Hypothesis 5.0 will drop Python 2 support as close
+    # to midnight as reasonably possible, without being early.  For details see:
+    # https://docs.travis-ci.com/user/customizing-the-build/#build-timeouts
+    if parse_release_file()[0] == "major":
+        at = datetime(2020, 1, 1)
```

Comment:
Why hardcode this to 2020 as opposed to using the current year? We're likely to want to leave this code in for next year.

Reply:
I expected that we'd get rid of it after this release, since dropping Python 2 is basically the only thing we want to release on a schedule.

If we want to keep it, I think the simplest way to use it is to bump the date each time, so it will be part of the code review.

Reply:
> 

I expected that we'd get rid of it after this release, since dropping Python 2 is basically the only thing we want to release on a schedule.

I figure if we're releasing that close to midnight on NYE we might as well align it with midnight - it's a nice touch, and if we're not then it has no impact.

> If we want to keep it, I think the simplest way to use it is to bump the date each time, so it will be part of the code review.

I don't understand how writing code that we have to remember to be update each time is supposed to be simpler than writing code that works automatically?

Reply:
Fair enough; I'll update it to use the end of the current year :+1: